### ISSUE TITLE
TPE using 1/1 trials with best loss higher

### ISSUE THREAD

**Issue by bandeep2000** [regel/loudml#114]
Using the latest version of loudml:

Seeing best loss > 4, before it was around 0.01, is this expected?
What does best loss mean, trying to look in the docs also

loudml train --from "now-6d" --to "now-2d" avg_cpu_idle1

INFO:hyperopt.tpe:TPE using 0 trials
INFO:hyperopt.tpe:tpe_transform took 0.002101 seconds
INFO:hyperopt.tpe:TPE using 1/1 trials with best loss 5.594314
INFO:hyperopt.tpe:tpe_transform took 0.002080 seconds
INFO:hyperopt.tpe:TPE using 2/2 trials with best loss 5.545225
INFO:hyperopt.tpe:tpe_transform took 0.002012 seconds
INFO:hyperopt.tpe:TPE using 3/3 trials with best loss 5.545225
INFO:hyperopt.tpe:tpe_transform took 0.002080 seconds
INFO:hyperopt.tpe:TPE using 4/4 trials with best loss 5.152984

Model details:
{
  "offset": "30",
  "features": [
          {
              "metric": "avg",
              "field": "usage_idle",
              "name": "mean_usage_system",
              "default": 0,
              "measurement": "cpu",
              "anomaly_type": "low_high",
               "match_all": [{"tag": "host", "value": "machine1"}]

          }
  ],
  "bucket_interval": "1m",
  "min_threshold": 75,
  "max_threshold": 95,
  "name": "avg_cpu_idle1",
  "default_datasource": "favorite_source",
  "interval": "1m",
  "max_evals": 5,
  "type": "donut",
  "span": 10,
  "forecast": 5
}

cat /etc/issue
Debian GNU/Linux 9 \n \l

Just installed loudml from  the latest version:

https://loudml.io/guide/en/loudml/reference/current/deb.html

**Comment by regel** [comment]
@bandeep2000 

I read in your log: 5.59, 5.54, 5.54, and then 5.15. The loss is decreasing: good news, this means the neural network converges and finds a better fit.

You will get minor changes and different output for each run. This is the normal behavior with Tensorflow: layers in the DNN are initialized with random initial states.

This is described here: https://keras.io/getting-started/faq/#how-can-i-obtain-reproducible-results-using-keras-during-development

If you want to get a reproducible output you will have to set environment variables before you run the CLI:
 CUDA_VISIBLE_DEVICES="" PYTHONHASHSEED=0 

I'm pushing a new patch on master today with the changes to support this.

Let me know if we can close this issue.

**Comment by regel** [comment]
Closing.
